src/BinSkim.Driver/RoslynAnalyzer/ILImporter.cs

```diff
@@ -0,0 +1,881 @@
+                boxedValue = value;
+                type = Int64Type;
+            }
+
+            Push(kind, new LiteralExpression(boxedValue, type));
+        }
+
+        private void ImportLoadFloat(double value)
+        {
+            Push(StackValueKind.Float, new LiteralExpression(value, DoubleType));
+        }
+
+        private void ImportLoadNull()
+        {
+            Push(StackValueKind.ObjRef, new LiteralExpression(null, ObjectType));
+        }
+
+        private void ImportReturn()
+        {
+            Append(new ReturnStatement(_method.ReturnsVoid ? null : Pop().Expression));
+        }
+
+        private void ImportFallthrough(BasicBlock next)
+        {
+            RequireEmptyStackForNow();
+            MarkBasicBlock(next);
+        }
+
+        private void ImportSwitchJump(int jmpBase, int[] jmpDelta, BasicBlock fallthrough)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportBranch(ILOpcode opcode, BasicBlock target, BasicBlock fallthrough)
+        {
+            StackValue left, right;
+
+            if (opcode == ILOpcode.brtrue || opcode == ILOpcode.brfalse)
+            {
+                left = Pop();
+                right = new StackValue(left.Kind, GetZeroLiteral(left.Kind));
+                opcode = opcode == ILOpcode.brtrue ? ILOpcode.bne_un : ILOpcode.beq;
+            }
+            else
+            {
+                right = Pop();
+                left = Pop();
+            }
+
+            Append(
+                new IfStatement(
+                    new BinaryOperatorExpression(
+                        GetBranchKind(opcode, GetStackKind(left.Kind, right.Kind)),
+                        left.Expression,
+                        right.Expression,
+                        BooleanType),
+                    new GoToStatement(
+                        GetOrCreateLabel(target))));
+
+            ImportFallthrough(fallthrough);
+            MarkBasicBlock(target);
+        }
+
+        private void ImportBinaryOperation(ILOpcode opcode)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportShiftOperation(ILOpcode opcode)
+        {
+            var right = Pop();
+            var left = Pop();
+
+            Push(
+                left.Kind,
+                new BinaryOperatorExpression(
+                    GetShiftKind(opcode),
+                    left.Expression,
+                    right.Expression,
+                    left.Expression.ResultType));
+        }
+
+        private void ImportCompareOperation(ILOpcode opcode)
+        {
+            throw new NotImplementedException();
+        }
+
+        private void ImportConvert(WellKnownType wellKnownType, bool checkOverflow, bool unsigned)
+        {
+            // unsigned argument deliberately unusued: it is captured in wellKnownType.
+            // FEEDBACK: How to represent checkOverflow = true?
```

Comment:
I'm not sure how important it its, but not having it means that it's not possible to write an IOperation analyzer that looks for integer overflow without being subject to false positives when the operations are already 'checked'.